lib/api/utilities/property_name_converter.rb

```diff
@@ -0,0 +1,70 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+module API
+  module Utilities
+    class PropertyNameConverter
+      class << self
+        WELL_KNOWN_ATTRIBUTE_CONVERSIONS = {
+          assigned_to: 'assignee',
+          fixed_version: 'version',
+          done_ratio: 'percentageDone',
+          estimated_hours: 'estimatedTime',
+          created_on: 'createdAt',
+          updated_on: 'updatedAt',
+          remaining_hours: 'remainingTime',
+          spent_hours: 'spentTime'
+        }
```

Comment:
Some of those are not really something we can "fix in the API", but we should fix in the underlying code: Most prominently `created_at` vs `created_on`.
In that specific case the API is simply more consistent than the existing code (the API **always** uses `createdAt`), however that does not mean it can't be fixed. It is just that probably nobody wants to write a database migration for the sake of consistency ^^

But I am all in to make the code more consistent and by that reduce the size of this list :)